pyccel/codegen/printing/pycode.py

```diff
@@ -229,7 +295,17 @@ def _print_Interface(self, expr):
         func = expr.functions[0]
         if not isinstance(func, FunctionAddress):
             func.rename(expr.name)
-        return self._print(func)
+        func_code = self._print(func)
+        _, body = func_code.split(':\n',1)
+        for func in expr.functions[1:]:
+            if not isinstance(func, FunctionAddress):
+                func.rename(expr.name)
+            i_func_code = self._print(func)
+            _, i_body = i_func_code.split(':\n',1)
+            if i_body != body:
+                warnings.warn(UserWarning("Generated code varies between interfaces but has not been printed. This Python code may produce unexpected results."))
+                return func_code
+        return func_code
```

Comment:
I have cleaned this function up a bit and added comments: 38086f94

I am checking that the generated code is the same. E.g. if we translate:
```python
@template('T', [float, int])
def f(a : 'T', b : 'T'):
    return a + b
```

then the Python printer will generate:
```python
def f(a : 'float', b : 'float'):
    return a + b
```
and
```python
def f(a : 'int', b : 'int'):
    return a + b
```
The body of these 2 functions is the same so we can print:
```python
def f(a : 'float', b : 'float'):
    return a + b
```
and know that it will work for the tests.

But if we translate:
```python
@template('T', [float, complex])
def f(a : 'T'):
    return a.imag
```
then we generate:
```python
def f(a : 'float', b : 'float'):
    return 0.0
```
and
```python
def f(a : 'complex', b : 'complex'):
    return a.imag
```
so a warning will be raised